tests/runfiles/linux.run

```diff
@@ -138,6 +138,10 @@ pre =
 post =
 tags = ['functional', 'largest_pool']
 
+[tests/functional/longname:Linux]
+tests = ['longname_001_pos', 'longname_002_pos', 'longname_003_pos']
+tags = ['functional', 'longname']
```

Comment:
Is there a reason we can't enable these on FreeBSD?

Reply:
I don't know if it would work on FreeBSD or not. I guess I can let buildbot to test it.

Reply:
Yes, let's see what happened on FreeBSD.

nit: I see there's also a [cstyle warning](https://github.com/openzfs/zfs/actions/runs/10190884366/job/28191292745?pr=15921) to sort out.